src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -458,60 +458,38 @@ public void CreateNewMicrobe()
 
-        // // organelleCount = 0;
-        // var previousMP = mutationPoints;
-        // // Copy current microbe to a new array
-        // array < PlacedOrganelle@> oldEditedMicrobeOrganelles = editedMicrobeOrganelles;
+        foreach (var organelle in editedMicrobeOrganelles)
+        {
+            oldEditedMicrobeOrganelles.Add(organelle);
+        }
 
-        // EditorAction@ action = EditorAction(0,
-        //     // redo
-        //     function(EditorAction@ action, MicrobeEditor@ editor){
-        //     // Delete the organelles (all except the nucleus) and set mutation points
-        //     // (just undoing and redoing the cost like other actions doesn't work in
-        //     // this case due to its nature)
-        //     editor.setMutationPoints(BASE_MUTATION_POINTS);
-        //     for (uint i = editor.editedMicrobeOrganelles.length() - 1; i > 0; --i)
-        //     {
-        //         const PlacedOrganelle@ organelle = editor.editedMicrobeOrganelles[i];
-        //         var hexes = organelle.organelle.getRotatedHexes(organelle.rotation);
-        //         for (uint c = 0; c < hexes.length(); ++c)
-        //         {
-        //             int posQ = int(hexes[c].X) + organelle.q;
-        //             int posR = int(hexes[c].Y) + organelle.r;
-        //             var organelleHere = OrganellePlacement::getOrganelleAt(
-        //                 editor.editedMicrobeOrganelles, Int2(posQ, posR));
-        //             if (organelleHere! is null)
-        //             {
-        //                OrganellePlacement::removeOrganelleAt(editor.editedMicrobeOrganelles,
-        //                     Int2(posQ, posR));
-        //             }
-        //
-        //         }
-        //     }
-        //
-        //     editor._onEditedCellChanged();
-        //
-        // },
-        //     function(EditorAction@ action, MicrobeEditor@ editor){
-        //     editor.editedMicrobeOrganelles.resize(0);
-        //     editor.setMutationPoints(int(action.data["previousMP"]));
-        //     // Load old microbe
-        //     array < PlacedOrganelle@> oldEditedMicrobeOrganelles =
-        //           cast<array<PlacedOrganelle@>> (action.data["oldEditedMicrobeOrganelles"]);
-        //     for (uint i = 0; i < oldEditedMicrobeOrganelles.length(); ++i)
-        //     {
-        //         editor.editedMicrobeOrganelles.insertLast(cast<PlacedOrganelle>(
-        //             oldEditedMicrobeOrganelles[i]));
-        //     }
-        //
-        //     editor._onEditedCellChanged();
-        // });
-        // @action.data["oldEditedMicrobeOrganelles"] = oldEditedMicrobeOrganelles;
-        // action.data["previousMP"] = previousMP;
-        // EnqueueAction(action);
+        var action = new EditorAction(this, 0,
+            redo =>
+            {
+                MutationPoints = Constants.BASE_MUTATION_POINTS;
+
+                for (int i = editedMicrobeOrganelles.Count - 1; i > 0; --i)
```

Comment:
I think that `RemoveAll` in editedMicrobeOrganelles should work here, would make this code more compact.

Reply:
Wouldn't that also remove the last organelle?

Reply:
Yeah, but didn't that also happen before? Anyway, I think that just adding a single cytoplasm after clearing is still clearer.